### ISSUE TITLE
Ask for some more details on experiments for reproducing the result.

### ISSUE THREAD

**Issue by AkiyaNguyen** [pingchuan/RD-Net#8]
Sincerely thank you for your contribution. 
I faced difficulty to reproduce the result reported in your paper, even when running with your original seed in your code. I have few questions to discuss more on your paper, please help me for this:
1. What's your exact version of Depth Anything model you used for creating depth images?
2. Where do you take the empirical results reported in table 2 on multiple semi-supevised methods? It seems like not all of those papers conduct experiments on polyp dataset. I guess that you took their repo and just add the dataset to run? Correct me if I understand something wrong.

Thank you, looking forward to hearing from you.

**Comment by pingchuan** [comment]
Thank you very much for your interest in our work! We are happy to clarify the questions you raised as follows:
1. Depth Anything Model Version
We used the officially released Depth-Anything-Large model, which has been explicitly stated in our paper.
2. Source of Results in Table 2 and Dataset Partitioning
To construct the semi-supervised training set, we followed the common practice in prior works by combining 900 training images from the Kvasir-SEG dataset and 550 training images from the CVC-ClinicDB dataset, resulting in a new training set of 1,450 images.
Since previous studies did not publicly disclose their exact data splits on polyp datasets, we randomly selected 10% of these 1,450 images as labeled samples, with the remaining 90% treated as unlabeled. All compared methods—including the semi-supervised approaches listed in Table 2—were re-implemented and evaluated under this identical data split (with a fixed random seed of 8888) on a unified experimental platform and under consistent settings to ensure a fair comparison.
For the hyperparameter settings and implementation details of our proposed method, RD-Net, please refer to the relevant sections in our paper.

**Comment by AkiyaNguyen** [comment]
<img width="508" height="349" alt="Image" src="https://github.com/user-attachments/assets/ed4b58d3-5232-42b0-98d9-41b5cf1f23fe" />

Thank you for your help. I follow almost strictly every single config in your code, except that I need to add code to save/load model (Save and load 2 students and teacher, 2 optimizer, 2 scheduler, epoch and global_iter) for multiple train sessions until reach 50000 iterations as mentioned in your paper. Another difference is that I use gpu t4x2 from kaggle to train (As I did some search, this gpu is not weaker or less efficient than yours). Sadly, after 2 runs (the first run I used the depth anything model v2 large, the second run is with the v1 large, same as yours), I still can not reach the record in your paper. The current result still remains much lower than yours. 

I can not deduce a possible cause for this phenomena. These above 2 mentioned difference could not make that huge difference in performance. Hope to get some help or advice from you.

**Comment by pingchuan** [comment]
Our publicly released code is a simplified version of the original implementation, designed to improve readability and usability; therefore, there may be minor discrepancies in certain implementation details. Nevertheless, the results reported in the paper are easy to reproduce.
Learning rate decay is scheduled based on scaled training epochs, where the scaling factor is determined by the labeled data ratio, specifically:
nEpoch_scaled = nEpoch × round((100 − labeled_perc) / labeled_perc).
This design compensates for the reduced number of labeled samples in semi-supervised training, thereby maintaining a reasonable number of optimization steps.
In addition, we are currently conducting a thorough review of the publicly released codebase to identify and fix any other potential issues.
If you are using our previously released training code, slightly adjusting the initial learning rate is usually sufficient to approximately reproduce the performance reported in the paper. Alternatively, you may also refer to the latest train.py.
Please note that minor variations in results may occur due to differences in experimental environments, such as hardware configurations, deep learning frameworks, or third-party library versions.
<img width="1048" height="192" alt="Image" src="https://github.com/user-attachments/assets/e0b604f1-ab35-4e4a-81d9-d43b0b436e0b" />

<img width="1051" height="275" alt="Image" src="https://github.com/user-attachments/assets/8e1c33a1-3a2e-4b8a-abf0-0b0da95a0bb1" />

<img width="1061" height="286" alt="Image" src="https://github.com/user-attachments/assets/86d664cf-a3d1-4c9f-9c90-725038c70815" />

Just now, we retrained the model using the same experimental settings as those reported in Table 2 of the paper, and evaluated it on the test set after every epoch for clear visualization. As shown in the figure, the model already achieves performance on the Kvasir-SEG test set close to that reported in the paper at epoch 75 (i.e., 75 × 72 = 5,400 iterations)—significantly fewer than the preset maximum of 50,000 iterations.
Moreover, around the middle of training (approximately at 25,000 iterations), the model has already converged and reaches the performance level reported in the paper.

**Comment by AkiyaNguyen** [comment]
Just want to clarify, based on what you discuss in paper, it should be args.total_iter (not args.nEpoch) right? Since I saw in the code that you step the scheduler after every iterations. I'm not sure if I misunderstand something.

<img width="967" height="197" alt="Image" src="https://github.com/user-attachments/assets/3326b088-462d-494d-8934-92590bf2939e" />

<img width="649" height="176" alt="Image" src="https://github.com/user-attachments/assets/d1af3539-7678-4842-818f-c70cdb79deee" />

**Comment by pingchuan** [comment]
Thank you very much for your careful observation!
In fact, our learning rate scheduling is based on scaled epochs, and during training, scheduler.step() is called once at the end of each epoch.
The earlier open-sourced simplified version of train.py did contain some oversights, leading to inconsistency between the scheduling logic and the actual experimental setup. We have now corrected this in the latest code.
In the paper, only the description regarding learning rate scheduling will be aligned with the updated train.py implementation, while the rest remains unchanged.
Thank you again for your valuable feedback—this is very important for our future work!

**Comment by AkiyaNguyen** [comment]
Sincerely thank you for your clarify. Could you please give me source on how you split train/test dataset and split label/unlabel data please? I would appreciate if you can help provide me the name list of your train and test dataset as well as name list of your labeled and unlabeled data for each dataset you conduct experiment on.  Additionally, For twoBatchStreamSampler, i am not sure if you set shuffe=True on any dataset. 

Btw, the formula for scheduler is inconsistent with what is written in paper right? i think it should be lambda e: pow(1.0 - (e / args.nEpoch), args.power) rather than lambda e: 1.0 - pow((e / args.nEpoch), args.power), as you mentioned in your paper, or is it an intentional modification?

Forward to hearing from you.
